hazelcast/src/main/java/com/hazelcast/instance/ProtocolType.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance;
+
+import com.hazelcast.nio.Protocols;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+
+// todo move to internal package com.hazelcast.cluster.internal.impl ?
```

Comment:
TODO

Reply:
+1, thanks for the nudge. Since `ProtocolType` is used publicly in `EndpointConfig`, it does not make sense to move it to an internal package, so it is maintained in a public package.